fixed the locators to select long name entities

sghai opened the pull request:
Here is one case to show the issue with long names.. When we use long names, UI truncates them.
for ex:
domain_name set as "longnaaaaaaaaaaaaaaaaaaaaaaaaaaame"
UI truncates it and list it as "longnaaaaa...."

Now when we assert the entity after creation, selenium doesn't find it, as name is already truncated. I've updated the locator and this should fix the test case.

```
[sghai@dhcp1xx-xx robottelo]$ nosetests  -c robottelo.properties  -m test_create_domain_2 /home/sghai/QE/robottelo/tests/foreman/ui/test_domain.py
/usr/lib/python2.7/site-packages/fauxfactory/__init__.py:737: Warning: The FauxFactory class is deprecated. Please use functions from the fauxfactory module instead.
  category=Warning)
@Test: Create a new domain ... 2015-01-07 18:37:28 - root - DEBUG - NoSuchElementException: Could not locate element //div[contains(@class, 'jnotify-notification-error')].
ok

----------------------------------------------------------------------
XML: foreman-results.xml
----------------------------------------------------------------------
Ran 1 test in 36.476s

```

Ichimonji10 commented:
I changed the title of this PR due to @sghai's comment:

> Please do not merge this PR. More fixes are coming...

elyezer commented:
Cool stuff

omaciel commented:
I took a quick look at the changes here and it looks like it would resolve the specific issue. To be clear, creating organizations or locations with long names still cause issues when using the web UI, but thankfully @elyezer has filed http://projects.theforeman.org/issues/8861

sghai commented:
@elyezer : thank you for filing that issue.. :+1:

sghai commented:
Guys another issue is:

When the length of string is greater than 30 char then UI truncates it with `...` so locator gets changed. I've updated the name of locator with `select_filetered_entity`. But when the length of string is `<= 30` then locator is different( the one we are already using).
I spent some time to make one common locator which can be used for both cases. But no luck. 

Solution:
So I updated the search function and first calculating the length of entity and accordingly calling the search fn with appropriate locator. 
This covers both cases and related tests are working fine for all data types.

kbidarkar commented:
@sghai, niice. ACK

Ichimonji10 commented:
ACK
